### PR TITLE
Filter messageReceived notification in adapter by threadId

### PR DESCRIPTION
**Pull request by AmyL219**
# What

Filter messageReceived notification in adapter by threadId

# Why

The composite is specifically designed for single-threaded use. To avoid confusion when a user participates in multiple threads, we filter messageReceived notifications by threadId.

# How Tested

Use same token in joining two different threads. Observed if the adapter only emit the event for the message specific to this chat thread

# Process & policy checklist

- [x] I have updated the project documentation to reflect my changes if necessary.
- [x] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

**Review by emlynmac — APPROVED** [review]

File: packages/react-composites/src/composites/ChatComposite/adapter/AzureCommunicationChatAdapter.ts [hunk 1]
```diff
@@ -369,6 +369,12 @@ export class AzureCommunicationChatAdapter implements ChatAdapter {
   }
 
   private messageReceivedListener(event: ChatMessageReceivedEvent): void {
+    const isCurrentChatAdapterThread = event.threadId === this.chatThreadClient.threadId;
+
+    if (!isCurrentChatAdapterThread) {
```
Comment (JoshuaLai) [inline packages/react-composites/src/composites/ChatComposite/adapter/AzureCommunicationChatAdapter.ts:374]: nipick: do we even need the `const` in this case? I guess it makes it slightly clearer and easier to read. 
```suggestion
    if (event.threadId !== this.chatThreadClient.threadId) {
```

**Review by JoshuaLai — APPROVED** [review]

File: packages/react-composites/src/composites/ChatComposite/adapter/AzureCommunicationChatAdapter.ts (hunk 1 above)
Comment (AmyL219) [reply packages/react-composites/src/composites/ChatComposite/adapter/AzureCommunicationChatAdapter.ts:374]: I think using a descriptive variable name here could improve readability.

**Comment by AmyL219** [comment]
Based on our previous discussion, we're treating this behavior update in the adapter as a bug fix. As this might require Contoso to modify their implementation code, let me know if there are any additional steps to consider during the release. cc @JoshuaLai

**Comment by JoshuaLai** [comment]
@AmyL219 lets just make a not of it. When we create the release notes for the release branch we will provide a more detailed description if need be.

**Pull request merged**